examples/nivisa_dmm_measurement/_visa_dmm.py

```diff
@@ -0,0 +1,141 @@
+"""Helper classes and functions for MeasurementLink NI-VISA examples."""
+
+import logging
+import pathlib
+import sys
+from enum import Enum
+from types import TracebackType
+from typing import (
+    Optional,
+    Type,
+)
+
+import pyvisa
+import pyvisa.resources
+import pyvisa.typing
+
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+
+# Pin map instrument type constants for NI Instrument Simulator v2.0
+INSTRUMENT_TYPE_DMM_SIMULATOR = "DigitalMultimeterSimulator"
+
+SIMULATION_YAML_PATH = pathlib.Path(__file__).resolve().parent / "_visa_dmm_sim.yaml"
+"""
+Path to a simulation YAML file that can be used to simulate an NI Instrument Simulator v2.0
+with pyvisa-sim.
+"""
+
+RESOLUTION_DIGITS_TO_VALUE = {"3.5": 0.001, "4.5": 0.0001, "5.5": 1e-5, "6.5": 1e-6}
+
+
+class Function(Enum):
+    """Function that represents the measurement type."""
+
+    DC_VOLTS = 0
+    AC_VOLTS = 1
+
+
+FUNCTION_TO_VALUE = {
+    Function.DC_VOLTS: "VOLT:DC",
+    Function.AC_VOLTS: "VOLT:AC",
+}
+
+
+class Session:
+    """An NI-VISA DMM session."""
+
+    def __init__(
+        self,
+        resource_name: str,
+        id_query: bool = True,
+        reset_device: bool = True,
+        use_simulation: bool = False,
+    ) -> None:
+        """Create a real or simulated VISA resource manager."""
+        visa_library = f"{SIMULATION_YAML_PATH}@sim" if use_simulation else ""
+        self._resource_manager = pyvisa.ResourceManager(visa_library)
```

Comment:
if possible retain the helper methods for create_visa_resource_manager and create_visa_session inside the session class itself as private and call from init method

Reply:
Retained both `_create_visa_resource_manager` and `_create_visa_session` methods outside the Session class similar to `_check_instrument_error` private method and updated the `__init__` to call the private helpers for creating sessions.